projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -588,9 +592,31 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseCompone
 
     private hg_verticalScrollHandler(event) {
         this.scrollTop = event.target.scrollTop;
+        this.hideOverlays();
+    }
+
+    public onContainerScrol() {
+        this.hideOverlays();
     }
 
     private hg_horizontalScrollHandler(event) {
         this.scrollLeft = event.target.scrollLeft;
+        this.hideOverlays();
+    }
+
+    private hideOverlays() {
+        const allOverlays = (this.overlayService as any)._overlayInfos;
+        const mainOverlays = allOverlays.filter((o) => o.settings.outlet === this.outletDirective);
+        if (mainOverlays.length > 0) {
+            mainOverlays.forEach(overlay => {
+                this.overlayService.hide(overlay.id);
+                // blur in case some editor somewhere decides to move focus back
+                this.overlayService.onClosed.pipe(
+                    filter(o => o.id === overlay.id),
+                    takeUntil(this.destroy$)).subscribe(() => {
+                        (document.activeElement as any).blur();
```

Comment:
@MayaKirova @ChronosSF We need to check whether calling blur in the destroy event of the igxFocus directive will work.

Reply:
@mpavlinov No. It does not.